### DISCUSSION TITLE
Edge Functions: DecompressionStream is not undefined

### DISCUSSION THREAD

**Discussion by jakeklassen** [supabase/supabase#28092] [Questions]
I just created a new project and a new edge function locally to test out streams.

This is a small function test to stream a gzip file, decompress, read, count and report the count in the response.

```ts
import "https://esm.sh/@supabase/functions-js/src/edge-runtime.d.ts";
import { TextLineStream } from "https://deno.land/std@0.224.0/streams/mod.ts";

Deno.serve(async (req) => {
  try {
    console.time("Download movie exports");
    const movieExportsResponse = await fetch(
      "https://files.tmdb.org/p/exports/movie_ids_07_01_2024.json.gz",
    );
    console.timeEnd("Download movie exports");

    const movieExportsCountStream = movieExportsResponse.body
      ?.pipeThrough(new DecompressionStream("gzip"))
      .pipeThrough(new TextDecoderStream())
      .pipeThrough(new TextLineStream());

    if (movieExportsCountStream == null) {
      throw new Error("Unable to get movie exports stream");
    }

    let movieExportCount = 0;

    for await (const _movieExport of movieExportsCountStream) {
      movieExportCount++;
    }

    console.log(`Movie export count: ${movieExportCount}`);

    return new Response(JSON.stringify({ movieExportCount }), {
      headers: { "content-type": "application/json" },
    });
  } catch (error) {
    console.error(error);
    return new Response(String(error?.message ?? error), { status: 500 });
  }
});
```

This is resulting in the error `DecompressionStream is not defined`. Is anyone else running into this? I can confirm this exact code works fine in a Deno project outside of supabase - it's where I am porting my code from.

This is my `pnpm supabase functions serve` output: `Using supabase-edge-runtime-1.54.10 (compatible with Deno v1.43.0)`

**I have also tested this in a cloud hosted supabase project, deployed the function, and same issue.**

**Comment by nyannyacha** [comment]
Hello @jakeklassen 😋

I just submitted [a PR](https://github.com/supabase/edge-runtime/pull/398) for this.
Thank you for reporting it!
